apiserver/internal/handlers/resources/resources.go

```diff
@@ -264,37 +276,38 @@ func (h *ResourceHandler) getUploadedResource(
 		resourceName:    res.Resource.Name,
 		size:            uReq.Size,
 		fingerprint:     uReq.Fingerprint,
+		pending:         pending,
 	}, nil
 }
 
 // getResourceUUID returns the resource uuid to match the new resource blob to.
-func getResourceUUID(
+func (h *ResourceHandler) getResourceUUID(
 	ctx context.Context,
 	resourceService ResourceService,
 	uReq api.UploadRequest,
-) (coreresource.UUID, error) {
+) (coreresource.UUID, bool, error) {
 	// If there is a valid PendingID in the request, the client has already
 	// setup the resource to expect a new upload, no need to do it again.
 	updatedResourceUUID, err := coreresource.ParseUUID(uReq.PendingID)
 	if err == nil {
-		return updatedResourceUUID, nil
+		return updatedResourceUUID, true, nil
```

Comment:
We've only parsed the UUID, should we not also see if this UUID refers to a resource?